sql-plugin/src/main/scala/com/nvidia/spark/rapids/limit.scala

```diff
@@ -129,7 +129,81 @@ case class GpuLocalLimitExec(limit: Int, child: SparkPlan) extends GpuBaseLimitE
  * Take the first `limit` elements of the child's single output partition.
  */
 case class GpuGlobalLimitExec(limit: Int, child: SparkPlan, offset: Int) extends GpuBaseLimitExec {
+  assert(limit >= 0 || (limit == -1 && offset > 0))
   override def requiredChildDistribution: List[Distribution] = AllTuples :: Nil
+
+  override def doExecuteColumnar(): RDD[ColumnarBatch]  = {
```

Comment:
Could we support offset by updating the code in `GpuBaseLimitExec`, instead of rewriting here ?
Then for `GpuGlobalLimitExec`, we pass through the offset parameter, but for `GpuLocalLimitExec`, force the offset being 0.